tests/application/job_service/test_job.py

```diff
@@ -15,7 +15,7 @@ class TestJob:
 
     @pytest.mark.asyncio
     async def test_if_service_raises_error_then_job_raises_error(self):
-        mock_apply_to_elster = MagicMock(side_effect=EricProcessNotSuccessful)
+        mock_apply_to_elster = MagicMock(side_effect=EricProcessNotSuccessful(610201215))
```

Comment:
Why did you add an error code here? And isn't calling the exception here changing how the test works? Because I think having the side_effect as a class is probably different than having an instance?

Reply:
I know why you added the code now. :D

Reply:
Because the retrying of jobs works like this: if within the job execution an exception is raised, then the job will be retried, otherwise it will be marked as finished. That is why I added this part within the exception-catching in the job.py:
` if entity.error_code in ERIC_ERRORS_WITH_RETRY:
raise
`
Therefore we need an specific error code passed in the EricProcessNotSuccessful, otherwise it will be UNKNOWN_ERROR, which is not in the list for a retry

Reply:
Iiinteresting..so you cant pass a class or an instance of the error you want to be raised. TIL. :) Thank you for explaining. 👍